mongo/session.go

```diff
@@ -365,39 +277,35 @@ func (s *sessionImpl) CommitTransaction(ctx context.Context) error {
 }
 
 // ClusterTime implements the Session interface.
-func (s *sessionImpl) ClusterTime() bson.Raw {
+func (s *Session) ClusterTime() bson.Raw {
 	return s.clientSession.ClusterTime
 }
 
 // AdvanceClusterTime implements the Session interface.
-func (s *sessionImpl) AdvanceClusterTime(d bson.Raw) error {
+func (s *Session) AdvanceClusterTime(d bson.Raw) error {
 	return s.clientSession.AdvanceClusterTime(d)
 }
 
 // OperationTime implements the Session interface.
-func (s *sessionImpl) OperationTime() *bson.Timestamp {
+func (s *Session) OperationTime() *bson.Timestamp {
 	return s.clientSession.OperationTime
 }
 
 // AdvanceOperationTime implements the Session interface.
-func (s *sessionImpl) AdvanceOperationTime(ts *bson.Timestamp) error {
+func (s *Session) AdvanceOperationTime(ts *bson.Timestamp) error {
 	return s.clientSession.AdvanceOperationTime(ts)
 }
 
 // Client implements the Session interface.
-func (s *sessionImpl) Client() *Client {
+func (s *Session) Client() *Client {
 	return s.client
 }
 
-// session implements the Session interface.
-func (*sessionImpl) session() {
-}
-
 // sessionFromContext checks for a sessionImpl in the argued context and returns the session if it
 // exists
 func sessionFromContext(ctx context.Context) *session.Client {
 	s := ctx.Value(sessionKey{})
-	if ses, ok := s.(*sessionImpl); ses != nil && ok {
+	if ses, ok := s.(*Session); ses != nil && ok {
 		return ses.clientSession
 	}
```

Comment:
Optional: This function should use `SessionFromContext` instead of using `sessionKey{}` directly.

E.g.
```go
func sessionFromContext(ctx context.Context) *session.Client {
	sess := SessionFromContext(ctx)
	if sess == nil {
		return nil
	}

	return sess.clientSession
}
```